eva/optimizer/plan_generator.py

```diff
@@ -86,8 +92,54 @@ def optimize(self, logical_plan: Operator):
         optimal_plan = self.build_optimal_physical_plan(root_grp_id, optimizer_context)
         return optimal_plan
 
+    # Disable exchange plan if there is a branch.
+    def post_process(self, physical_plan: AbstractPlan):
+        # Detect whether there is a branch.
+        is_branch, is_create_mat = False, False
+        for plan_node in physical_plan.walk():
+            if len(plan_node.children) > 1:
+                is_branch = True
+                break
+            if isinstance(plan_node, CreateMaterializedViewPlan):
+                is_create_mat = True
+
+        # Replace exchange plan.
+        if is_branch or is_create_mat:
+
+            def _recursive_strip_exchange(plan: AbstractPlan, is_top: bool = False):
+                children = []
+                for child_plan in plan.children:
+                    return_child = _recursive_strip_exchange(child_plan)
+                    if isinstance(return_child, List):
+                        children += return_child
+                    else:
+                        children += [return_child]
+
+                plan.clear_children()
+                for child in children:
+                    plan.append_child(child)
+
+                if isinstance(plan, ExchangePlan):
+                    if is_top:
```

Comment:
I think this code piece can be simplified. How about you assert on `is_top` and `len(plan.children) == 1` and always return `plan.children`.  This is also simplify the above check on `List`.

Reply:
Is this a temporary work around? I feel We should be able to support branch and createMat. But I agree it is not priority.